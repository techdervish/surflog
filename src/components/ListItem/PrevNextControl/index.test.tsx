import {
  expectWrappersTreeToMatchSnapshot,
  createShallowWrapper,
} from 'utils/testHelpers';
import Component from './index';

const createWrapper = createShallowWrapper(Component);

describe('PrevNextControl component', () => {
  const mockOnBack = jest.fn();
  const mockOnNext = jest.fn();

  const wrapper = createWrapper({
    onClickBack: mockOnBack,
    onClickNext: mockOnNext,
    total: 6,
    current: 2,
  });

  it('renders the expected snapshot', () => {
    expectWrappersTreeToMatchSnapshot(wrapper);
  });

  it('should call onClickBack prop on click', () => {
    wrapper
      .find('a')
      .first()
      .simulate('click');
    expect(mockOnBack).toBeCalled();
  });

  it('should call onClickNext prop on click', () => {
    wrapper
      .find('a')
      .last()
      .simulate('click');
    expect(mockOnNext).toBeCalled();
  });

  it('do not display prev if current is zero', () => {
    const wrapper = createWrapper({
      total: 6,
      current: 0,
    });
    expect(
      wrapper
        .find('a')
        .last()
        .text(),
    ).toEqual('Next > ');
  });

  it('do not display next if current is total', () => {
    const wrapper = createWrapper({
      total: 6,
      current: 6,
    });
    expect(
      wrapper
        .find('a')
        .last()
        .text(),
    ).toEqual(' < Prev | ');
  });
});
