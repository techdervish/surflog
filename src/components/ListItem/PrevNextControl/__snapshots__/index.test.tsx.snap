// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrevNextControl component renders the expected snapshot 1`] = `
<div
  className="control bottom-0 right-0 flex absolute h1 f7 justify-center"
  style={
    Object {
      "background": "#01032480",
      "minWidth": 136,
    }
  }
>
  <div>
    (3 of 6)
     |
     
    <a
      onClick={[MockFunction]}
    >
       &lt;
       Prev 
      <span>
        | 
      </span>
    </a>
    <a
      onClick={[MockFunction]}
    >
      Next &gt; 
    </a>
  </div>
</div>
`;
